virtool/subtractions/db.py

```diff
@@ -181,3 +183,31 @@ def lookup_nested_subtractions(
             }
         },
     ]
+
+
+def subtraction_processor(data: list, query: dict) -> dict:
+    try:
+        data = data[0]
+    except IndexError:
+        raise ResourceNotFoundError()
+
+    data["documents"] = [
+            base_processor(document) for document in data["documents"]
+    ]
+
+    try:
+        data["page"] = int(query["page"])
+    except (KeyError, ValueError):
+        data["page"] = 1
+
+    per_page = None
+    try:
+        per_page = int(query["per_page"])
+    except (KeyError, ValueError):
+        per_page = 25
+    finally:
+        data["per_page"] = per_page
+
+    data["page_count"] = int(math.ceil(data["found_count"] / per_page))
+
+    return data
```

Comment:
Not a bad idea in principle to separate out complicated logic, but as written this function is a bit confused. There are core logic problems. Most problematically it appears to be both getting parameters needed to do pagination (`per_page`, `page`), while also calculating values like `page_count` which would be calculated after running the aggregation pipeline.

`per_page` and `page` are essential to have before the pipeline so that the correct documents can be returned. (per page indicates how many documents there are per page, while page is which page the user is on) The two parameters combined can be used to tell the pipeline which documents should be returned. (imagine the case where there are 100 subtractions, which ones should be returned if the user requests 25 documents per page and the third page)